### PR TITLE
First architecture diagram

### PR DESCRIPTION
**Pull request by jschaul**

### CODE REVIEW

**Review by kkohbrok — APPROVED** [review]
Looks good! I'll take the token for Authentication, Authorization and Discovery after this PR got landed.

File: src/understand/federation/architecture.rst [hunk 1]
```diff
@@ -4,3 +4,66 @@ Architecture and Network
...
+Per-request Authorization
+^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+TODO.
+
+
+..
+  paths to images are currently listed at the end of the file. If you prefer to specify them directly in the paragraph they are used, that is also fine.
+
+.. |flow-exact-handle-search| image:: img/exact-handle-search.png
```
Comment (akshaymankar) [inline src/understand/federation/architecture.rst:68]: I thought we were going to use the python diagrams thing?

File: src/understand/federation/img/exact-handle-search.txt [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+
+federator@B -> brig@B: Request("/handle/alice")
+
+brig@B -> federator@B: Response
+
+federator@B -> federator@A: Response
+
+federator@A -> brig@A: Response: UserProfile of Alice
+
+
```
Comment (akshaymankar) [inline src/understand/federation/img/exact-handle-search.txt:25]: We should add a note saying the contents of the request are not accurate or make the contents accurate.

**Review by akshaymankar — CHANGES_REQUESTED** [review]
Let's add a note regarding accuracy of the endpoints.

File: src/understand/federation/img/exact-handle-search.txt (hunk 2 above)
Comment (jschaul) [reply src/understand/federation/img/exact-handle-search.txt]: good point, done.

File: src/understand/federation/architecture.rst (hunk 1 above)
Comment (jschaul) [reply src/understand/federation/architecture.rst:103]: At some point, yes, makes sense to explore. Not today though, scope creep for this PR :)

**Review by akshaymankar — APPROVED** [review]
:guitar:

**Pull request merged**